src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+    private final Random random;
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+        this.random = new Random();
+
```

Comment:
```suggestion
       colorSupplier = new ColorSupplier();
       random = new Random();
```
or pass both of them